### ISSUE TITLE
Doesn't work with Swift

### ISSUE THREAD

**Issue by michaelforrest** [matthewcheok/Realm-JSON#37]
Currently this doesn't work with Swift because when the class is matched by string it doesn't take into account Swift's module prefix. It doesn't necessarily seem like a complicated fix to me - just wondering if it's on your radar?
Thanks

**Comment by jbejar** [comment]
Has anyone got this to work with Swift?

**Comment by matthewcheok** [comment]
Is this still an issue?

**Comment by nanoxd** [comment]
Yes, with RealmSwift the syntax has changed. Will there ever be support for RealmSwift in this project? I feel like this should keep using the Objective-C API with a different version of this using the Swift API.

**Comment by orkenstein** [comment]
`Realm-swift` is based on `Object` (does not look like a good name for class) subclassed from `RLMObjectBase`. I don't think it's a big deal to transfer decoding/coding to `RLMObjectBase`. Right, @matthewcheok?

**Comment by matthewcheok** [comment]
I'll try to get some time to take a closer look. Meanwhile PRs are welcome!

**Comment by orkenstein** [comment]
@matthewcheok, I'm currently experimenting with swift support, but got stuck with initWithDictionary. I moved extension to `RLMObjectBase`, and auto-generated swift code looks like:
```
extension RLMObjectBase {
  .....
    public init!(JSONDictionary dictionary: [NSObject : AnyObject]!)
    public func JSONDictionary() -> [NSObject : AnyObject]!
    
   ....
}
```

Which is seems ok to me. But *Realm* `Object` does not inherit `init!(JSONDictionary dictionary: [NSObject : AnyObject]!)` by some reason. It's not available.
Any suggestions?

**Comment by orkenstein** [comment]
After some experiments, it does not look to me as so simple )
It's possible to move basic functionality to `RLMObjectBase`, but we need to rewrite methods like `createOrUpdateInRealm` for **Realm's** `Object`. Any suggestions?

**Comment by rad182** [comment]
+1 for RealmSwift support

**Comment by justjoeyuk** [comment]
Needs swift supports

**Comment by deivuh** [comment]
+1 on Swift support. 

Tried out other libraries. Some of them are either not compatible with Realm or the current version of RealmSwift.

**Comment by Changsung** [comment]
+1 on Swift support.

**Comment by justjoeyuk** [comment]
Guys, I recommend ObjectMapper as a solution (https://github.com/Hearst-DD/ObjectMapper) it works great for me.

**Comment by oparinov** [comment]
@justjoeyuk how do you handle with List<T> while using ObjectMapper?

**Comment by Changsung** [comment]
I made RealmSwift-JSON(https://github.com/Changsung/RealmSwift-JSON) that imitates Realm-JSON. 
I hope it would be a solution which you are looking for
